ledger/complete/mtrie/trie/trie.go

```diff
@@ -195,6 +195,33 @@ func valueSizes(sizes []int, paths []ledger.Path, head *node.Node) {
 	}
 }
 
+// ReadSinglePayload reads and returns a payload for a single path.
+func (mt *MTrie) ReadSinglePayload(path ledger.Path) *ledger.Payload {
+	return readSinglePayload(path, mt.root)
+}
+
+// readSinglePayload reads and returns a payload for a single path in subtree with `head` as root node.
+func readSinglePayload(path ledger.Path, head *node.Node) *ledger.Payload {
+	pathBytes := path[:]
+
+	// Traverse nodes following the path until a leaf node or nil node is reached.
+	for !head.IsLeaf() {
+		depth := ledger.NodeMaxHeight - head.Height() // distance to the tree root
+		bit := bitutils.ReadBit(pathBytes, depth)
+		if bit == 0 {
+			head = head.LeftChild()
+		} else {
+			head = head.RightChild()
+		}
+	}
```

Comment:
maybe sliiightly faster :) 

```suggestion
        depth := ledger.NodeMaxHeight - head.Height() // distance to the tree root
	for !head.IsLeaf() {
		bit := bitutils.ReadBit(pathBytes, depth)
		if bit == 0 {
			head = head.LeftChild()
		} else {
			head = head.RightChild()
		}
		depth++
	}
```